src/rust/src/metrics_ffi.rs

```diff
@@ -137,104 +131,78 @@ pub extern "C" fn metrics_key(
         let labels = stringify(labels);
         let values = stringify(values);
 
-        let labels: Vec<_> = labels
-            .into_iter()
-            .zip(values.into_iter())
-            .map(|(name, value)| Label::new(name, value))
-            .collect();
+        let labels = labels.into_iter().zip(values.into_iter()).collect();
 
-        Box::into_raw(Box::new(FfiKey {
-            inner: Key::from_parts(name, labels),
-        }))
+        Box::into_raw(Box::new(FfiKey { name, labels }))
     }
 }
 
 #[no_mangle]
 pub extern "C" fn metrics_static_increment_counter(callsite: *const FfiCallsite, value: u64) {
-    if let Some(recorder) = try_recorder() {
-        let callsite = unsafe { callsite.as_ref().unwrap() };
-        recorder.increment_counter(&callsite.key, value);
-    }
+    let callsite = unsafe { callsite.as_ref().unwrap() };
+    counter!(callsite.0.name, value, &callsite.0.labels);
 }
 
 #[no_mangle]
 pub extern "C" fn metrics_increment_counter(key: *mut FfiKey, value: u64) {
-    if let Some(recorder) = try_recorder() {
-        if !key.is_null() {
-            let key = unsafe { Box::from_raw(key) };
-            recorder.increment_counter(&key.inner, value);
-        }
+    if !key.is_null() {
+        let key = unsafe { Box::from_raw(key) };
+        counter!(key.name, value, &key.labels);
     }
 }
 
 #[no_mangle]
 pub extern "C" fn metrics_static_update_gauge(callsite: *const FfiCallsite, value: c_double) {
-    if let Some(recorder) = try_recorder() {
-        let callsite = unsafe { callsite.as_ref().unwrap() };
-        recorder.update_gauge(&callsite.key, GaugeValue::Absolute(value));
-    }
+    let callsite = unsafe { callsite.as_ref().unwrap() };
+    gauge!(callsite.0.name, value as f64, &callsite.0.labels);
 }
 
 #[no_mangle]
 pub extern "C" fn metrics_update_gauge(key: *mut FfiKey, value: c_double) {
-    if let Some(recorder) = try_recorder() {
-        if !key.is_null() {
-            let key = unsafe { Box::from_raw(key) };
-            recorder.update_gauge(&key.inner, GaugeValue::Absolute(value));
-        }
+    if !key.is_null() {
+        let key = unsafe { Box::from_raw(key) };
+        gauge!(key.name, value as f64, &key.labels);
     }
 }
 
 #[no_mangle]
 pub extern "C" fn metrics_static_increment_gauge(callsite: *const FfiCallsite, value: c_double) {
-    if let Some(recorder) = try_recorder() {
-        let callsite = unsafe { callsite.as_ref().unwrap() };
-        recorder.update_gauge(&callsite.key, GaugeValue::Increment(value));
-    }
+    let callsite = unsafe { callsite.as_ref().unwrap() };
+    increment_gauge!(callsite.0.name, value as f64, &callsite.0.labels);
```

Comment:
In a pairing with @daira, I went through the new `metrics-macros` impl and determined that we could still do what was done previously; we just need to have slightly different calls in each FFI method. e.g.
```rust
    if let Some(recorder) = try_recorder() {
        let callsite = unsafe { callsite.as_ref().unwrap() };
        let handle = recorder.register_gauge(&callsite.key);
        handle.increment(value);
    }
```